dessia_common/datatools/modeling.py

```diff
@@ -0,0 +1,734 @@
+"""
+Tools and base classes for machine learning methods.
+
+"""
+from typing import List, Dict, Any
+
+import numpy as npy
+from sklearn import preprocessing, linear_model, ensemble, tree, svm, neural_network
+
+from dessia_common.core import DessiaObject
+
+
+# ======================================================================================================================
+#                                                     S C A L E R S
+# ======================================================================================================================
+class BaseScaler(DessiaObject):
+    _rebuild_attributes = []
+
+    def __init__(self, name: str = ''):
+        DessiaObject.__init__(self, name=name)
+
+    @classmethod
+    def _skl_class(cls):
+        raise NotImplementedError('Method _skl_class not implemented for BaseScaler. Please use children.')
+
+    def _call_skl_scaler(self):
+        return self._skl_class()()
+
+    def instantiate_skl(self):
+        scaler = self._call_skl_scaler()
+        for attr in self._rebuild_attributes:
+            setattr(scaler, attr, getattr(self, attr))
+        return scaler
+
+    @classmethod
+    def instantiate_dessia(cls, scaler, name: str = ''):
+        kwargs_dict = {'name': name}
```

Comment:
Consider naming only "kwargs" as "kwargs_dict" seems rendundant. That's actually a pleonasm :)

Reply:
Done everywhere :)